### DISCUSSION TITLE
[GENERAL] at a high level, how does AC work?

### DISCUSSION THREAD

**Discussion by Frichetten** [azerothcore/azerothcore-wotlk#3970] [General]
Hi all, hope I posted this in the correct thread. 

I stood up an AC server earlier today, and frankly I'm stunned. A few Docker commands and a few tweaks and suddenly I'm playing Wotlk circa 2009. From what little I've played it appears like everything is working pretty flawlessly. 

I'm aware this is a dumb question, but how is this all working? As in, was this made entirely by monitoring the network traffic between the client and server and then building a server to mimic that traffic?

A better question, in this recreation, how does it know where to put NPCs? Did someone meticulously examine the retail version and recreate all of their paths and where they stand?

**Comment by FrancescoBorzi** [comment]
Thanks a lot @Frichetten for your interest in AzerothCore! I reopen the question because I think it's worth answering it.

> I stood up an AC server earlier today, and frankly I'm stunned. A few Docker commands and a few tweaks and suddenly I'm playing Wotlk circa 2009. From what little I've played it appears like everything is working pretty flawlessly.

I'm really glad to hear it, thanks for sharing your experience!

> I'm aware this is a dumb question, but how is this all working? As in, was this made entirely by monitoring the network traffic between the client and server and then building a server to mimic that traffic?
> A better question, in this recreation, how does it know where to put NPCs? Did someone meticulously examine the retail version and recreate all of their paths and where they stand?

The work behind this project has been HUGE and it's the outcome of years of development within different projects (MaNGOS, TrinityCore, SunwellCore, AzerothCore) from hundreds of contributions. I find it to be one of the best example of what can be achieved with open source and international collaboration. I believe even the best developers couldn't do this job alone. All small details, all the single NPCs where added by volunteers with time.

See: https://github.com/azerothcore/azerothcore-wotlk#authors--contributors

**Comment by BarbzYHOOL** [comment]
But yes, some of the work consisted in monitoring the network traffic, aka sniffing. Never touched this but it usually happens on every new expansion

**Comment by Kaev** [comment]
There were many different ways used to create this experience. As BarbzYHOOL and FrancescoBorzi said, many things were done by hand and many things were done by packet sniffing.
Some things were extracted out of the client (e.g. extracting item data or npc informations out of the client database files, the client also caches e.g. the item informations as soon as you click on a linked item in chat so the cache files helped aswell).
Some things were gathered by the whole WoW community (e.g. a good base for the item drops were the informations that you can find on WoWHead) and some things were generated by things we figured out over time and by reverse engineering the game client (e.g. by figuring out the WoW map file format and generating movement maps out of these informations).

I'm following and supporting the development of WoW private servers since very early TBC and i'm proud of the quality we already achieved today. It's not perfect but it's damn good for something we had to figure out by ourselves over all these years.
Kudos to all related people that supported the development, the developers and the game over all these years.
